appinterface/projection/crossfire/crossfire.go

```diff
@@ -195,21 +195,19 @@ func (projection *Crossfire) handleBlockCreatedEvent(
 
 	// increment current phase block number
 	if blockTime.AfterOrEqual(projection.phaseOneStartTime) && blockTime.Before(projection.phaseTwoStartTime) {
-		projection.profile("begin incrementing phase 1 block count")
 		err := crossfireChainStatsView.IncrementOne(constants.PHASE_1_BLOCK_COUNT)
 		if err != nil {
 			return fmt.Errorf("[Crossfire] error increment phase1 block count")
 		}
-		projection.profile("end incrementing phase 1 block count")
 	} else if blockTime.AfterOrEqual(projection.phaseTwoStartTime) && blockTime.Before(projection.phaseThreeStartTime) {
-		projection.profile("begin incrementing phase 2 block count")
 		err := crossfireChainStatsView.IncrementOne(constants.PHASE_2_BLOCK_COUNT)
 		if err != nil {
 			return fmt.Errorf("[Crossfire] error increment phase2 block count")
 		}
-		projection.profile("end incrementing phase 2 block count")
 	} else if blockTime.AfterOrEqual(projection.phaseThreeStartTime) && blockTime.Before(projection.competitionEndTime) {
 		// check the keep active task, throttling with every 10 blocks
+		// keep active task is only counting commitments during phase 2, so checking it in phase 3
+		// guarantees correctness
 		if blockHeight%10 == 0 {
```

Comment:
For the last `%10` Number of the block, It may not check to keep active?
For eg: 
Let's say: The `blockheight` at `competitionEndTime` happens to be `99`
Now our logic will last execute `checkTaskKeepActive ` at `90` .
Which means for `(99-90 = 9)` Blocks we would have missed the task completion checking.

Shall we remove `blockHeight%10 == 0` check to mitigate this?
@calvinaco

Reply:
The keep active task concerns of up time during phase 2 only; while this check is done during phase 3. So at the time it is executed the keep alive data are fixed.

So there shouldn't be problem. In fact we may further extend to longer time interval for optimizing efficiency.

Reply:
Ahh. True.. . .The underlying function only deals with phase 2 values. Cool